lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -250,6 +263,8 @@ def create_launch_config(connection, module):
             if bdm.ebs is not None:
                 result['block_device_mappings'][-1]['ebs'] = dict(snapshot_id=bdm.ebs.snapshot_id, volume_size=bdm.ebs.volume_size)
 
+    if launch_configs[0].user_data != "":
+        result['user_data'] = "hidden" # Otherwise, we dump binary to the user's terminal
```

Comment:
One `-v` is enough to get module results these days.

A reasonable tradeoff might be

```
if user_data_path:
    result['user_data'] = 'hidden'
```

as the compelling reason to use `user_data_path` over `user_data` with a lookup is to allow zipped input. More sophisticated options might be to check if the data is compressed, or just binary (see e.g. http://stackoverflow.com/a/7392391/3538079) and hide if so.